File: benchmark/perl/hanoi.pl
use Time::HiRes qw(time);
sub hanoi {
    my $n = shift;
    return 1 if $n == 1;
    return hanoi($n - 1) + 1 + hanoi($n - 1);
}
my $start = time();
my $moves = hanoi(20);
my $end = time();
my $ms = ($end - $start) * 1000;
print "Moves: $moves
";
printf "Time: %.0fms
", $ms;

File: benchmark/python/hanoi.py
import time

def hanoi(n):
    if n == 1:
        return 1
    return hanoi(n - 1) + 1 + hanoi(n - 1)

start = time.time() * 1000
moves = hanoi(20)
end = time.time() * 1000
print(f"Moves: {moves}")
print(f"Time: {end - start:.0f}ms")

